src/types.ts

```diff
@@ -509,3 +549,38 @@ function tagType(tag: Tag): Type<any> {
 		throw new TypeError(`unsupported tag ${tag}`);
 	}
 }
+
+function createSubType(tag: Tag, type: Type<number>) {
+	return {
+		enc(buf: WriteBuffer, v: number): void {
+			buf.putUi8(tag);
+
+			switch(tag) {
+			case Tag.Nil: break;
+
+			// signed int types
+			case Tag.Int8: buf.putI8(v); break;
+			case Tag.Int16: buf.putI16(v); break;
+			case Tag.Int32: buf.putI32(v); break;
+			case Tag.Int64: buf.putI64(v); break;
+
+			// unsigned int types
+			case Tag.Uint8: buf.putUi8(v); break;
+			case Tag.Uint16: buf.putUi16(v); break;
+			case Tag.Uint32: buf.putUi32(v); break;
+			case Tag.Uint64: buf.putUi64(v); break;
+
+			// float types
+			case Tag.Float32: buf.putF32(v); break;
+			case Tag.Float64: buf.putF64(v); break;
+
+			default:
+				typeError(tag, "");
+			}
+		},
+
+		dec(buf: ReadBuffer): number {
+			return type.dec(buf);
```

Comment:
Can you please provide some more details how the suggested fixed-size types should behave?
For example, the `Int8` type uses the same decoding function as the `Int` type. This can result in an value that does not fit in 8 bits. What is the advantage of the `Int8` type if you can't be sure you always get an 8-bit integer?
Maybe you can elaborate a little bit more on this.

Reply:
my situation is the 90Hz time value is Uint32, even overflowed is ok, but the length need to be Uint32, even value is small

Reply:
For your situation the overflow might be ok. But maybe we can develop a more general specification what the semantics of the fixed-size types should be. I see an advantage in the writing part. These types ensure, that you always write the specified tag and integer width into the binary stream. This could help in supporting existing protocols. Shouldn't it then be the same for the reading part?
So the question for me currently is: Should an `IntX` value should always be translated from and to the [MessagePack "int X" wire format](https://github.com/msgpack/msgpack/blob/master/spec.md#int-format-family)?

Reply:
i think
Int means min is Int8 max Int64
Int8 means min is Int16 max Int64
Int16 means min is Int16 max Int64
Int32 means min is Int32 max Int64

maybe should add suffix "unsafe" or better one ?
Int8Unsafe means force Int8
Int16Unsafe means force Int16
Int32Unsafe means force Int32

Reply:
Ok, then let's come to the following decission: An `IntX` type should always be encoded to and decoded from an _X_ bit value ignoring overflow scenarios. This means that we always use and assume the binary format of `int X`. A `UintX` type should behave equivalently with unsigned values.

Reply:
yes